### PR TITLE
Touchup of Plug tutorial

### PR DESCRIPTION
**Pull request by moxley**
## Changes

* Add more explanation for those not knowledgeable in OTP.
* Some touch-up for consistency
* Rename "Installation" section to "Prerequisites", which is more accurate.
* Add section for starting new project.
* More explanation about the roles and arguments of a Plug.

On my last PR for this lesson (https://github.com/doomspork/elixir-school/pull/861), I never opened an issue for translations. Should I open one now, or after this is merged?

### CODE REVIEW

**Comment by doomspork** [comment]
@moxley thank you for these changes!  I'll look over them tonight.  We must missed making an issue, oops!  If you wouldn't mind doing that, that'd be great!

Can you please update the version at the top of the file to be `1.1.0`?  We're using versioning to notify readers when translations may be out-of-date.

**Comment by moxley** [comment]
@doomspork: Lesson version updated and translation issue created: #1043

**Comment by doomspork** [comment]
Sorry for the delays @moxley, I promise to get to this this week 😀

File: lessons/specifics/plug.md [hunk 1]
```diff
@@ -83,10 +100,13 @@ end
 
 This supervises Cowboy, and in turn, supervises our `HelloWorldPlug`.
 
-Now, the `application` portion of `mix.exs` needs two things:
+In the `Plug.Adapters.Cowboy.child_spec/4` call, the third argument will be passed to `Example.HelloWorldPlug.init/1`.
+
+We're not finished yet. Open `mix.exs` again, and find the `applications` function.
```
Comment (doomspork) [inline lessons/specifics/plug.md]: Are most people using 1.4 these days?  If so then mentioning `applications` could be confusing, we might want to update these lessons with the 1.4 changes.  That can be a separate PR though.

File: lessons/specifics/plug.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-version: 1.0.0
+version: 1.1.0
```
Comment (doomspork) [inline lessons/specifics/plug.md]: Thank you for bumping this ❤️

**Comment by doomspork** [comment]
Thank you @moxley!
